lib/rust/ensogl/core/src/debug/stats.rs

```diff
@@ -103,3 +247,43 @@ macro_rules! if_compiled_with_stats {
         {}
     };
 }
+
+
+
+// ========================
+// === ValueAccumulator ===
+// ========================
+
+#[derive(Debug, Default)]
+struct ValueAccumulator<T> {
+    pub min: T,
+    pub max: T,
+    pub sum: f64,
+}
+
+impl<T: Min + Max + PartialOrd + cast::AsPrimitive<f64> + Copy> ValueAccumulator<T> {
+    fn new(v: T) -> Self {
+        Self { min: v, max: v, sum: v.as_() }
+    }
+
+    fn push(&mut self, v: T) {
+        self.min = min(self.min, v);
+        self.max = max(self.max, v);
+        self.sum += v.as_();
+    }
+}
+
+
+
+// ====================
+// === ValueSummary ===
+// ====================
+
+/// Summarized data observed for a single stat over a number of GUI rendering frames.
+#[derive(Clone, Debug, Serialize, Deserialize)]
```

Comment:
Why not Copy? Please do not change the code and close this comment – it's an important design decision – why the decision was done not to include Copy here?

Reply:
This was due to me still having holes in my Rust knowledge - in particular, until late in this PR's development, I missed the existence of the `Copy` trait (and once I discovered it, didn't realize I should review the PR carefully and check if I missed adding it somewhere).